src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,301 @@
+    Returns:
+        List[Tuple[float, float]]: a list of the densified points
+    """
+    src = rasterio.open(href)
+
+    # create datamask
+    if no_data is None:
+        no_data = src.nodata
+
+    scaled_out_shape = (int(src.shape[0] / scale), int(src.shape[1] / scale))
+    arr = src.read(1, out_shape=scaled_out_shape)
+    data_val = 1 if no_data != 1 else 0
+
+    if no_data is not None:
+        arr[numpy.where(arr != no_data)] = data_val
+    else:
+        arr.fill(data_val)
+
+    shapes_transform = src.transform * Affine.scale(scale)
+
+    data_polygons = [
+        shape(polygon_dict)
+        for polygon_dict, region_value in rasterio.features.shapes(
+            arr, transform=shapes_transform
+        )
+        if region_value == data_val
+    ]
+
+    if not data_polygons:
+        return None
+    elif len(data_polygons) == 1:
+        polygon = data_polygons[0]
+    else:
+        polygon = MultiPolygon(data_polygons).convex_hull
+
+    polygon = densify_reproject_simplify(
+        polygon,
+        src.crs,
+        densification_factor=densification_factor,
+        precision=precision,
+        simplify_tolerance=simplify_tolerance,
+    )
+
+    import json
+
+    print(json.dumps(mapping(polygon)))
+
+    return mapping(polygon)  # type: ignore
+
+
+def densify_reproject_simplify(
+    polygon: Polygon,
+    crs: CRS,
+    *,
+    densification_factor: Optional[int] = None,
+    precision: int = DEFAULT_PRECISION,
+    simplify_tolerance: Optional[float] = None,
+) -> Polygon:
+    """
+        From the input Polygon, densifies the polygon, reprojects it to EPSG:4326, and then
+        simplifies the resulting polygon.
+
+        See :py:meth:`update_geometry_from_asset_footprint` for more details about densification
+        and simplification.
+
+     Args:
+        polygon (Polygon): The input Polygon
+        crs (CRS): The CRS of the input Polygon
+        densification_factor (Optional[int]): The factor by which to increase point density
+            within the polygon. A factor of 2 would double the density of points (placing one
+            new point between each existing pair of points), a factor of 3 would place two points
+            between each point, etc.
+        precision (int): The number of decimal places to include in the coordinates for the
+            reprojected geometry. Defaults to 3 decimal places.
+        simplify_tolerance (Optional[float]): All points in the simplified object will be within
+            the tolerance distance of the original geometry, in degrees.
+
+    Returns:
+        Polygon: the reprojected Polygon
+    """
+    if densification_factor is not None:
+        polygon = Polygon(_densify(polygon.exterior.coords, densification_factor))
+
+    polygon = shape(transform_geom(crs, "EPSG:4326", polygon, precision=precision))
+
+    if simplify_tolerance is not None:
+        polygon = polygon.simplify(
+            tolerance=simplify_tolerance, preserve_topology=False
```

Comment:
I deleted a prior comment where I recommended setting `preserve_topology` to `False` since my understanding of topology preservation only applies to multiple poygons (prevent overlaps and gaps from adjacent polygons). I deleted the comment because a bit more research revealed that some simplification algorithms do produce different polygon depending on a `preserve_topology`-like setting, and I don't know the characteristics of shapely's algorithms. Perhaps just remove the explicit setting and let shapely's default of `True` for `preserve_topology` to hold.

Reply:
so this is odd, because the GEOS documentation says that it doesn't apply to polygons, but I got different results with True and False for only one of our footprints. The test that checks precision setting simplified to a polygon that had a line in it like, but with False it removed that line, which is better, but surprising because it wasn't supposed to affect polygons! Maybe this is an outlier case? 

|-----|
|-----|
|-----|
|--------------